### PR TITLE
PT-2588: Allow to use 'status' term in filter expression

### PR DESCRIPTION
**Pull request by tatarincev**
Use the folowing filter syntax to search variations
```json
  filter: "status:hidden is:variation"
```

### CODE REVIEW

**Comment by mvktsk** [comment]
Review task https://virtocommerce.atlassian.net/browse/VP-7485 has been created

File: src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Index/IndexSearchRequestBuilder.cs [hunk 1]
```diff
@@ -73,26 +73,21 @@ public IndexSearchRequestBuilder AddObjectIds(IEnumerable<string> ids)
             }
             return this;
         }
+        public IndexSearchRequestBuilder AddTermsIfNotExists(IEnumerable<string> terms)
```
Comment (Vectorfield4) [inline src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Index/IndexSearchRequestBuilder.cs:76]:
```suggestion

        public IndexSearchRequestBuilder AddTermsIfNotExists(IEnumerable<string> terms)
```

File: src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Index/IndexSearchRequestBuilder.cs [hunk 2]
```diff
@@ -307,9 +302,35 @@ public virtual SearchRequest Build()
...
         }
 
-        private void AddFiltersToSearchRequest(IFilter[] filters)
+        private void AddFiltersToSearchRequest(IFilter[] filters, bool skipIfExists = false)
+        {
+            var childFilters = ((AndFilter)SearchRequest.Filter).ChildFilters;
+            //Skip adding duplicate filters
+            if (skipIfExists)
+            {                
+                var existsFiltersNames = childFilters.OfType<INamedFilter>().Select(x=>x.FieldName).Distinct().ToArray();
```
Comment (Vectorfield4) [inline src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Index/IndexSearchRequestBuilder.cs:311]:
```suggestion
                var existsFiltersNames = childFilters.OfType<INamedFilter>().Select(x => x.FieldName).Distinct();
```

File: src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Index/IndexSearchRequestBuilder.cs [hunk 3]
```diff
@@ -307,9 +302,35 @@ public virtual SearchRequest Build()
...
+                filters = filters.Where(x=> !(x is INamedFilter namedFilter && existsFiltersNames.Contains(namedFilter.FieldName, StringComparer.InvariantCultureIgnoreCase))).ToArray();
```
Comment (Vectorfield4) [inline src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Index/IndexSearchRequestBuilder.cs:312]:
```suggestion
                filters = filters.Where(x=> !(x is INamedFilter namedFilter && existsFiltersNames.Contains(namedFilter.FieldName, StringComparer.InvariantCultureIgnoreCase)));
```

**Review by Vectorfield4 — APPROVED** [review]

**Pull request merged**